src/mixins/nft.js

```diff
@@ -517,109 +515,95 @@ export default {
-
+      let historyMap = null;
       if (this.nftIsWritingNFT) {
-        promises.push(
-          getNFTHistoryDataMap({
-            axios: this.$api,
-            classId: this.classId,
-            nftId: this.nftId,
-          })
-        );
+        historyMap = await getNFTHistoryDataMap({
+          axios: this.$api,
+          classId: this.classId,
+          nftId: this.nftId,
+        });
       }
 
-      const eventMap = new Map();
-      history.forEach(event => {
-        eventMap.set(getEventKey(event), event);
-      });
-
-      const [priceMap, historyMap] = await Promise.all(promises);
-      priceMap.forEach((price, key) => {
-        if (eventMap.has(key)) {
-          eventMap.get(key).price = price;
-        }
-      });
-      if (historyMap) {
-        historyMap.forEach((data, key) => {
-          if (eventMap.has(key)) {
-            const {
-              classId,
-              nftId,
-              grantTxHash,
-              granterMemo,
-              granterWallet,
-              price,
-              timestamp,
-            } = data;
-            if (grantTxHash && granterMemo) {
-              const e = {
+      const addGrantEventForWNFTAndUpdateUserInfo = history => {
+        if (this.nftIsWritingNFT) {
+          const historyWithGrant = [];
+          history.forEach(event => {
+            const key = getEventKey(event);
+            if (historyMap.has(key)) {
+              const {
                 classId,
                 nftId,
-                fromWallet: granterWallet,
-                event: 'grant',
-                memo: granterMemo,
-                txHash: grantTxHash,
-                timestamp: timestamp + 1,
-              };
-              history.push(e);
-              eventMap.set(grantTxHash, e);
-              eventMap.get(key).granterMemo = granterMemo;
+                grantTxHash,
+                granterMemo,
+                granterWallet,
+                timestamp,
+              } = historyMap.get(key);
+              if (grantTxHash && granterMemo) {
+                const e = {
+                  classId,
+                  nftId,
+                  fromWallet: granterWallet,
+                  event: 'grant',
+                  memo: granterMemo,
+                  txHash: grantTxHash,
+                  timestamp: timestamp + 1,
+                };
+                historyWithGrant.push(e);
+                // eslint-disable-next-line no-param-reassign
+                event.granterMemo = granterMemo;
+              }
             }
-            eventMap.get(key).price = price;
-          }
-        });
-      }
-      history.sort((a, b) => b.timestamp - a.timestamp);
+            // add original event after grant event for time reverse order
+            historyWithGrant.push(event);
+          });
+          this.NFTHistory.push(...historyWithGrant);
+        } else {
+          this.NFTHistory.push(...history);
```

Comment:
When do we reset `this.NFTHistory`?

Reply:
Probably safer to reset it before pushing. Nice caught.

Reply:
Would be easier to read if you return `...historyWithGrant` or `...history` in this function
then explicitly call push when calling i.e. `this.NFTHistory.push(addGrantEventForWNFTAndUpdateUserInfo(...))`

Reply:
Would be even cleaner if we completely avoid using `this`, and use function param instead in `addGrantEventForWNFTAndUpdateUserInfo`

Reply:
updated, see if it's cleaner